src/xdist/scheduler/load.py

```diff
@@ -265,10 +271,10 @@ def schedule(self):
             # how many items per node do we have about?
             items_per_node = len(self.collection) // len(self.node2pending)
             # take a fraction of tests for initial distribution
-            node_chunksize = max(items_per_node // 4, 2)
+            node_chunksize = min(items_per_node // 4, self.maxschedchunk)
```

Comment:
For clarity let's keep all the steps to compute `node_chunksize` together:

```suggestion
            node_chunksize = min(items_per_node // 4, self.maxschedchunk)
            node_chunksize = max(node_chunksize, 2)
```